[OpenShift] Remove ca_certs, deprecate validate_cert, fix unreleased regression (#640)
jupyterhub/oauthenticator: docs/source/reference/changelog.md

  reading and adding entries must now use set logic and not list logic.
- [Google] Authentication state's `google_groups` is now a set, not a list.
- [CILogon] {attr}`.CILogonOAuthenticator.allowed_idps` is now required config,
  and `shown_idps`, `username_claim`, `additional_username_claims` were removed.
- [Okpy] The public functions `OkpyOAuthenticator.get_auth_request` and
  `OkpyOAuthenticator.get_user_info_request` were removed.

> - [OpenShift] The config `ca_certs` was removed but
>   {attr}`.OpenShiftOAuthenticator.http_request_kwargs` can still be configured
>   with a `ca_certs` key for the same result. OpenShift's default `ca_certs`

Reviewer: ```suggestion
- [OpenShift] The config `ca_certs` was removed. Use
  {attr}`.OAuthenticator.http_request_kwargs`
  with a `ca_certs` key instead. OpenShift's default `ca_certs`
```
